### PR TITLE
Add Undo and Redo

### LINKED ISSUE #45: Add Undo/Redo to Editor
__Is your feature request related to a problem? Please describe__

Currently, there is no undo/redo in the form editor.

__Describe the solution you'd like__

A command stack should be added to the form editor to add undo/redo. I'd argue that the form viewer doesn't need this feature since we do not want to mess with native undo/redo when using a form.

### PR DESCRIPTION
**Pull request by philippfromme**
* add and use destroy API
* add command stack, editor actions and keyboard binding services
* all modeling operations are executed as commands
* add test helpers

Closes #45

### CODE REVIEW

File: packages/form-js-editor/src/FormEditor.js [hunk 1]
```diff
@@ -87,6 +87,19 @@ export default class FormEditor {
     }
   }
 
+  destroy() {
+    this.get('eventBus').fire('form.destroy');
+
+    const container = this._container,
```
Comment (nikku) [inline packages/form-js-editor/src/FormEditor.js:96]: Why not use `detach` here? It should do the same thing, right?

File: packages/form-js-viewer/src/Form.js [hunk 2]
```diff
@@ -93,6 +93,19 @@ export default class Form {
     }
   }
 
+  destroy() {
+    this.get('eventBus').fire('form.destroy');
+
+    const container = this._container,
```
Comment (nikku) [inline packages/form-js-viewer/src/Form.js:99]: Why not use `detach` here? It should do the same thing, right?

**Review by nikku — CHANGES_REQUESTED** [review]
Works on my machine. Two small comments inlined.

File: packages/form-js-viewer/src/Form.js (hunk 2 above)
Comment (philippfromme) [reply packages/form-js-viewer/src/Form.js:99]: Cf. https://github.com/bpmn-io/form-js/pull/78#discussion_r648149434.

Comment (nikku) [reply packages/form-js-viewer/src/Form.js:99]: The alternative to duplication could be to be able to detach without emitting [here](https://github.com/bpmn-io/form-js/blob/master/packages/form-js-editor/src/FormEditor.js#L153).

```
this.detach(false);
```

Comment (nikku) [reply packages/form-js-viewer/src/Form.js:99]: If we wanted to we could even move the core detach functionality into its own api, i.e expose the `noDetach` arg on a `_detach` utility only

Comment (philippfromme) [reply packages/form-js-viewer/src/Form.js:99]: Yeah, I though about moving the duplicate code to a `_detach` method and using it in both places but firing the event only in `detach`. Will add a commit with a proposal.

Comment (philippfromme) [reply packages/form-js-viewer/src/Form.js]: Done (cf. https://github.com/bpmn-io/form-js/pull/78/commits/e672778fa2169fd8fc0f3765708b02b16ac96330).

File: packages/form-js-editor/src/FormEditor.js [hunk 3]
```diff
@@ -151,14 +160,23 @@ export default class FormEditor {
   }
 
   detach() {
+    this._detach();
```
Comment (barmac) [inline packages/form-js-editor/src/FormEditor.js:163]: How about if we don't use any flags and emit the event where we want?

```suggestion
    this._emit('detach');
    this._detach();
```

File: packages/form-js-editor/src/FormEditor.js [hunk 4]
```diff
@@ -151,14 +160,23 @@ export default class FormEditor {
...
           parentNode = container.parentNode;
 
     if (!parentNode) {
       return;
     }
 
-    this._emit('detach');
+    if (emit) {
+      this._emit('detach');
+    }
```
Comment (barmac) [inline packages/form-js-editor/src/FormEditor.js:179]: supplement to https://github.com/bpmn-io/form-js/pull/78/files#r648281710

```suggestion
```

File: packages/form-js-editor/src/FormEditor.js (hunk 3 above)
Comment (philippfromme) [reply packages/form-js-editor/src/FormEditor.js:163]: The behavior would be different. The event would be fired even if there wasn't a detach.

File: packages/form-js-viewer/src/Form.js (hunk 2 above)
Comment (nikku) [reply packages/form-js-viewer/src/Form.js]: Works for me. Works for you, too @barmac?

**Review by barmac — APPROVED** [review]

**Pull request merged**